tests/genai/judges/test_make_judge.py

```diff
@@ -366,3 +367,133 @@ def mock_invoke(model_uri, prompt, assessment_name):
+
+    retrieved_scorer = store.get_scorer(experiment, "test_judge", version)
+    assert retrieved_scorer is not None
+    assert isinstance(retrieved_scorer, InstructionsJudge)
+    assert retrieved_scorer.name == "test_judge"
+    assert retrieved_scorer.instructions == original_instructions
+    assert retrieved_scorer.model == "openai:/gpt-4"
+    assert retrieved_scorer.template_variables == {"outputs"}
+    assert retrieved_scorer.description == original_description
+    assert "Instructions-based judge: test_judge" in retrieved_scorer.description
+    assert original_instructions in retrieved_scorer.description
+
+    deserialized = Scorer.model_validate(serialized)
+    assert isinstance(deserialized, InstructionsJudge)
+    assert deserialized.name == judge.name
+    assert deserialized.instructions == original_instructions
+    assert deserialized.model == judge.model
+    assert deserialized.template_variables == {"outputs"}
+    assert deserialized.description == original_description
+
+    v2_instructions = "Evaluate if the output {{outputs}} is professional, formal, and concise."
+    judge_v2 = make_judge(
+        name="test_judge",
+        instructions=v2_instructions,
+        model="openai:/gpt-4o",
+    )
+    version2 = store.register_scorer(experiment, judge_v2)
+    assert version2 == 2
+
+    versions = store.list_scorer_versions(experiment, "test_judge")
+    assert len(versions) == 2
+
+    v1_scorer, v1_num = versions[0]
+    assert v1_num == 1
+    assert isinstance(v1_scorer, InstructionsJudge)
+    assert v1_scorer.instructions == original_instructions
+    assert v1_scorer.model == "openai:/gpt-4"
+
+    v2_scorer, v2_num = versions[1]
+    assert v2_num == 2
+    assert isinstance(v2_scorer, InstructionsJudge)
+    assert v2_scorer.instructions == v2_instructions
+    assert v2_scorer.model == "openai:/gpt-4o"
+
+    latest = store.get_scorer(experiment, "test_judge")
+    assert isinstance(latest, InstructionsJudge)
+    assert latest.instructions == v2_instructions
+    assert latest.model == "openai:/gpt-4o"
+    assert "Instructions-based judge: test_judge" in latest.description
+    assert v2_instructions in latest.description
+
+
+def test_judge_registration_preserves_custom_variables(tmp_path, mock_invoke_judge_model):
+    tracking_uri = "sqlite:///{}".format(tmp_path.joinpath("mlflow.db"))
+    mlflow.set_tracking_uri(tracking_uri)
+
+    experiment = mlflow.create_experiment("test_custom_vars")
+
+    instructions_with_custom = (
+        "Check if {{query}} is answered correctly by {{response}} "
+        "according to {{criteria}} with {{threshold}} accuracy"
+    )
+    judge = make_judge(
+        name="custom_judge",
+        instructions=instructions_with_custom,
+        model="openai:/gpt-4",
+    )
+
+    assert judge.template_variables == {"query", "response", "criteria", "threshold"}
+    assert judge._custom_template_variables == {"query", "response", "criteria", "threshold"}
```

Comment:
Is the asserting on this private property necessary? Seems like the previous assert is sufficent